src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketGitSCMBuilder.java

```diff
@@ -104,8 +104,9 @@ public BitbucketGitSCMBuilder(@NonNull BitbucketSCMSource scmSource, @NonNull SC
                 String branchName = ((PullRequestSCMHead) head).getBranchName();
                 withRefSpec("+refs/heads/" + branchName + ":refs/remotes/@{remote}/" + head.getName());
             } else {
-                String pullId = ((PullRequestSCMHead) head).getId();
-                withRefSpec("+refs/pull-requests/" + pullId + "/from:refs/remotes/@{remote}/" + head.getName());
+                // https://community.atlassian.com/t5/Bitbucket-questions/Difference-of-refs-pull-requests-lt-ID-gt-merge-and-refs-pull/qaq-p/772142
+                String branchName = ((PullRequestSCMHead) head).getBranchName();
+                withRefSpec("+refs/heads/" + branchName + ":refs/remotes/@{remote}/" + branchName);
```

Comment:
Not sure how this is supposed to handle PRs where source branch is in a fork?
Probably fine to do a branch lookup first, but maintain the `refs/pull-requests/x/from` as a fallback?

Reply:
shouldn't the `@{remote}` be the fork? 🤔

Reply:
`@{remote}` is only used in the "receiving" part of the refspec, after the colon, which just indicates how it's gonna store that ref "locally"; that's irrelevant with where it's supposed to find the source branch. Also if we were on the fork already, we wouldn't have to use those pull-requests refs in the first place. :)

Besides navigating through, as `#withRefSpec` indicates, `@{remote}` is supposed to be processed for substitution by `#remote()` before use, which is only ever gonna return a fixed remote for the current scm source, as far as I can tell.

Reply:
I havn't tested with forks, because we don't use them. Maybe i can try it.

Reply:
We've just tested plugin version `2.7.1-rc629.f242e60a14b` with our setup which uses forks, and indeed it fails at this step as the branches can not be found on the origin